### PR TITLE
Add PowerDNS Recursor input plugin

### PR DESCRIPTION
**Pull request by dupondje**
Added a PowerDNS Recursor input plugin.

The PowerDNS plugin does not work for the recursor, as it as some differences.
Not only the command to get the stats is different, but also the socket type (datagram socket for the recursor).

As I also had to add some additional options, it looks better to me to have it in a separate plugin.

Please review the code if it can be merged. If things need to get changed, let me know!
I also get a strange error on make test ... so might need some assistance there!

### CODE REVIEW

**Comment by dupondje** [comment]
Adjusted the requested changes.

That random came from the powerdns input plugin by the way:
https://github.com/influxdata/telegraf/blob/master/plugins/inputs/powerdns/powerdns_test.go#L76

So might want to change it there also :)

**Comment by dupondje** [comment]
And if merged, https://github.com/influxdata/telegraf/issues/812 can be closed!

**Comment by glinton** [comment]
@dupondje Can you investigate what might be taking so long in the tests? They are failing circleci, I thought it was due to the `rand` things, but even without them, it seems to be taking too long.
Thanks

**Comment by dupondje** [comment]
Test was fixed! :)

It was caused by the fact that with Unixgram sockets, you need to write the message to the remote socket, and not to the socket the message was read from.
So the Read succeeded, but the Write wasn't received by the plugin, so it was hanging forever ...

**Comment by dupondje** [comment]
@danielnelson: Any changes required to get it merged?

**Comment by dupondje** [comment]
What are we waiting for? :) Would like to get it merged.

**Comment by dupondje** [comment]
Any update for this? Would be nice to get it merged!

File: plugins/inputs/powerdns_recursor/README.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  # An array of sockets to gather stats about.
+  # Specify a path to unix socket.
+  #
+  # If no servers are specified, then '/var/run/pdns_recursor.controlsocket' is used as the path.
+  unix_sockets = ["/var/run/pdns_recursor.controlsocket"]
+
+  # Socket for Receive
+  unix_socket_recv_path = "/tmp/"
+  # Socket permissions
+  unix_socket_perm = 0666
```
Comment (danielnelson) [inline plugins/inputs/powerdns_recursor/README.md]: Can you rename these to `socket_dir` and `socket_mode`?

File: plugins/inputs/powerdns_recursor/powerdns_recursor.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+	for _, metric := range s[:len(s)-1] {
+		m := strings.Split(metric, "\t")
+		if len(m) < 2 {
+			continue
+		}
+
+		i, err := strconv.ParseInt(m[1], 10, 64)
+		if err != nil {
+			log.Printf("E! powerdns_recursor: Error parsing integer for metric [%s]: %s",
```
Comment (danielnelson) [inline plugins/inputs/powerdns_recursor/powerdns_recursor.go]: Can you update this to the latest style:
```
log.Printf("E! [inputs.powerdns_recursor] Error parsing integer for metric [%s] %v", metric, err)
```

File: plugins/inputs/powerdns_recursor/powerdns_recursor.go [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+var defaultTimeout = 5 * time.Second
+
+var sampleConfig = `
+  ## An array of sockets to gather stats about.
+  ## Specify a path to unix socket.
+  unix_sockets = ["/var/run/pdns_recursor.controlsocket"]
+
+  ## Socket for Receive
+  #unix_socket_recv_path = "/tmp/"
```
[the remark this answers is not in the document]
Comment (dupondje) [reply plugins/inputs/powerdns_recursor/powerdns_recursor.go]: Telegraf is not running with PrivateTmp by default?
If somebody changes it, yea then this breaks indeed.

Agree that /var/run/telegraf would be a better path, but needs to be created then indeed.

[the remark this answers is not in the document]
Comment (dupondje) [reply plugins/inputs/powerdns_recursor/powerdns_recursor.go]: Ah pdns-recursor runs indeed with PrivateTmp, so can't connect to /tmp indeed.
What about just put it inside /var/run?

[the remark this answers is not in the document]
Comment (dupondje) [reply plugins/inputs/powerdns_recursor/powerdns_recursor.go]: Changed to /var/run/ in the latest code now.

**Comment by glinton** [comment]
Can you update your branch with master please?

**Comment by dupondje** [comment]
Updated :)

**Comment by danielnelson** [comment]
@dupondje I think we just need the tests fixed and we should be good to merge.

**Comment by dupondje** [comment]
Should be good now!

**Pull request merged**